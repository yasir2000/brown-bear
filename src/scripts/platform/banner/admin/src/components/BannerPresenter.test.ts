/*
 * Copyright (c) Enalean, 2020 - Present. All Rights Reserved.
 *
 *  This file is a part of Tuleap.
 *
 *  Tuleap is free software; you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation; either version 2 of the License, or
 *  (at your option) any later version.
 *
 *  Tuleap is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Tuleap. If not, see <http://www.gnu.org/licenses/>.
 */

import { shallowMount } from "@vue/test-utils";
import BannerPresenter from "./BannerPresenter.vue";
import { createPlatformBannerAdminLocalVue } from "../helpers/local-vue-for-tests";

describe("BannerPresenter", () => {
    it("displays the banner and a checked switch if banner is set", async () => {
        const banner_message = "<b>My banner content</b>";

        const wrapper = shallowMount(BannerPresenter, {
            localVue: await createPlatformBannerAdminLocalVue(),
            propsData: {
                message: banner_message,
                importance: "critical",
                expiration_date: "",
                loading: false,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("displays a default message and an unchecked switch if banner is not set", async () => {
        const wrapper = shallowMount(BannerPresenter, {
            localVue: await createPlatformBannerAdminLocalVue(),
            propsData: {
                message: "",
                importance: "critical",
                expiration_date: "",
                loading: false,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("sets 'activated' to false on save-banner if the switch is clicked when banner is set", async () => {
        const wrapper = shallowMount(BannerPresenter, {
            localVue: await createPlatformBannerAdminLocalVue(),
            propsData: {
                message: "some message",
                importance: "critical",
                expiration_date: "",
                loading: false,
            },
        });

        const emitSpy = jest.spyOn(wrapper.vm, "$emit");

        wrapper.get("input").setChecked(false);
        wrapper.get("button").trigger("click");
        expect(emitSpy).toHaveBeenCalledWith("save-banner", {
            message: "some message",
            importance: "critical",
            expiration_date: "",
            activated: false,
        });
    });

    it("disables the whole form when on loading state", async () => {
        const wrapper = shallowMount(BannerPresenter, {
            localVue: await createPlatformBannerAdminLocalVue(),
            propsData: {
                message: "some message",
                importance: "critical",
                expiration_date: "",
                loading: true,
            },
        });

        expect(wrapper.element).toMatchSnapshot();
    });

    it("emits a save-banner event with the message on click on the save button", async () => {
        const wrapper = shallowMount(BannerPresenter, {
            localVue: await createPlatformBannerAdminLocalVue(),
            propsData: {
                message: "somme message",
                importance: "critical",
                expiration_date: "",
                loading: false,
            },
        });

        const emitSpy = jest.spyOn(wrapper.vm, "$emit");
        const updated_message = "new message";
        const updated_importance = "standard";

        wrapper.setData({
            current_message: updated_message,
            current_importance: updated_importance,
        });

        wrapper.get("button").trigger("click");
        expect(emitSpy).toHaveBeenCalledWith("save-banner", {
            message: updated_message,
            importance: updated_importance,
            expiration_date: "",
            activated: true,
        });
    });

    it("does not trigger a save-banner event if the user gives an empty message and banner has not been deactivated", async () => {
        const wrapper = shallowMount(BannerPresenter, {
            localVue: await createPlatformBannerAdminLocalVue(),
            propsData: {
                message: "some message",
                importance: "critical",
                expiration_date: "",
                loading: false,
            },
        });

        const emitSpy = jest.spyOn(wrapper.vm, "$emit");

        wrapper.setData({ current_message: "" });

        wrapper.get("button").trigger("click");
        expect(emitSpy).not.toHaveBeenCalled();
    });
});
