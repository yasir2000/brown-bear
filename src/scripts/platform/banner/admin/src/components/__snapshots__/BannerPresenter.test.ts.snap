// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BannerPresenter disables the whole form when on loading state 1`] = `
<div>
  <div
    class="tlp-form-element tlp-form-element-disabled"
  >
    <label
      class="tlp-label tlp-checkbox"
    >
      <input
        type="checkbox"
      />
       
      <translate-stub
        tag="span"
      >
        Activate the banner on the whole platform
      </translate-stub>
    </label>
  </div>
   
  <div>
    <div
      class="tlp-form-element siteadmin-platform-banner-importance"
    >
      <label
        class="tlp-label"
      >
        <translate-stub
          tag="span"
        >
          Importance
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fas fa-asterisk"
        />
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-standard"
          type="radio"
          value="standard"
        />
         
        <translate-stub
          class="tlp-text-info"
          tag="span"
        >
          Standard
        </translate-stub>
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-warning"
          type="radio"
          value="warning"
        />
         
        <translate-stub
          class="tlp-text-warning"
          tag="span"
        >
          Warning
        </translate-stub>
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-critical"
          type="radio"
          value="critical"
        />
         
        <translate-stub
          class="tlp-text-danger"
          tag="span"
        >
          Critical
        </translate-stub>
      </label>
    </div>
     
    <expiration-date-banner-input-stub
      value=""
    />
     
    <div
      class="tlp-form-element tlp-form-element-disabled"
    >
      <label
        class="tlp-label"
        for="description"
      >
        <translate-stub
          tag="span"
        >
          Message
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fas fa-asterisk"
        />
      </label>
       
      <textarea
        class="tlp-textarea"
        id="description"
        name="description"
        placeholder="Choose a banner message"
        style="visibility: hidden;"
      />
       
      <p
        class="tlp-text-muted"
        data-current-language="en_US"
        data-msgid="Your message will be condensed to one line"
      >
        Your message will be condensed to one line
      </p>
    </div>
  </div>
   
  <div
    class="tlp-pane-section-submit"
  >
    <button
      class="tlp-button-primary"
      data-tlp-tooltip="Message is mandatory"
      disabled="disabled"
      type="button"
    >
      <i
        class="tlp-button-icon fas fa-fw fa-spin fa-circle-notch"
      />
       
      <span
        data-current-language="en_US"
        data-msgid="Save the configuration"
      >
        Save the configuration
      </span>
    </button>
  </div>
</div>
`;

exports[`BannerPresenter displays a default message and an unchecked switch if banner is not set 1`] = `
<div>
  <div
    class="tlp-form-element"
  >
    <label
      class="tlp-label tlp-checkbox"
    >
      <input
        type="checkbox"
      />
       
      <translate-stub
        tag="span"
      >
        Activate the banner on the whole platform
      </translate-stub>
    </label>
  </div>
   
  <div
    style="display: none;"
  >
    <div
      class="tlp-form-element siteadmin-platform-banner-importance"
    >
      <label
        class="tlp-label"
      >
        <translate-stub
          tag="span"
        >
          Importance
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fas fa-asterisk"
        />
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-standard"
          type="radio"
          value="standard"
        />
         
        <translate-stub
          class="tlp-text-info"
          tag="span"
        >
          Standard
        </translate-stub>
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-warning"
          type="radio"
          value="warning"
        />
         
        <translate-stub
          class="tlp-text-warning"
          tag="span"
        >
          Warning
        </translate-stub>
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-critical"
          type="radio"
          value="critical"
        />
         
        <translate-stub
          class="tlp-text-danger"
          tag="span"
        >
          Critical
        </translate-stub>
      </label>
    </div>
     
    <expiration-date-banner-input-stub
      value=""
    />
     
    <div
      class="tlp-form-element"
    >
      <label
        class="tlp-label"
        for="description"
      >
        <translate-stub
          tag="span"
        >
          Message
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fas fa-asterisk"
        />
      </label>
       
      <textarea
        class="tlp-textarea"
        id="description"
        name="description"
        placeholder="Choose a banner message"
        style="visibility: hidden;"
      />
       
      <p
        class="tlp-text-muted"
        data-current-language="en_US"
        data-msgid="Your message will be condensed to one line"
      >
        Your message will be condensed to one line
      </p>
    </div>
  </div>
   
  <div
    class="tlp-pane-section-submit"
  >
    <button
      class="tlp-button-primary"
      data-tlp-tooltip="Message is mandatory"
      type="button"
    >
      <i
        class="tlp-button-icon fas fa-save"
      />
       
      <span
        data-current-language="en_US"
        data-msgid="Save the configuration"
      >
        Save the configuration
      </span>
    </button>
  </div>
</div>
`;

exports[`BannerPresenter displays the banner and a checked switch if banner is set 1`] = `
<div>
  <div
    class="tlp-form-element"
  >
    <label
      class="tlp-label tlp-checkbox"
    >
      <input
        type="checkbox"
      />
       
      <translate-stub
        tag="span"
      >
        Activate the banner on the whole platform
      </translate-stub>
    </label>
  </div>
   
  <div>
    <div
      class="tlp-form-element siteadmin-platform-banner-importance"
    >
      <label
        class="tlp-label"
      >
        <translate-stub
          tag="span"
        >
          Importance
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fas fa-asterisk"
        />
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-standard"
          type="radio"
          value="standard"
        />
         
        <translate-stub
          class="tlp-text-info"
          tag="span"
        >
          Standard
        </translate-stub>
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-warning"
          type="radio"
          value="warning"
        />
         
        <translate-stub
          class="tlp-text-warning"
          tag="span"
        >
          Warning
        </translate-stub>
      </label>
       
      <label
        class="tlp-label tlp-radio"
      >
        <input
          class="siteadmin-platform-banner-importance-critical"
          type="radio"
          value="critical"
        />
         
        <translate-stub
          class="tlp-text-danger"
          tag="span"
        >
          Critical
        </translate-stub>
      </label>
    </div>
     
    <expiration-date-banner-input-stub
      value=""
    />
     
    <div
      class="tlp-form-element"
    >
      <label
        class="tlp-label"
        for="description"
      >
        <translate-stub
          tag="span"
        >
          Message
        </translate-stub>
         
        <i
          aria-hidden="true"
          class="fas fa-asterisk"
        />
      </label>
       
      <textarea
        class="tlp-textarea"
        id="description"
        name="description"
        placeholder="Choose a banner message"
        style="visibility: hidden;"
      />
       
      <p
        class="tlp-text-muted"
        data-current-language="en_US"
        data-msgid="Your message will be condensed to one line"
      >
        Your message will be condensed to one line
      </p>
    </div>
  </div>
   
  <div
    class="tlp-pane-section-submit"
  >
    <button
      class="tlp-button-primary"
      data-tlp-tooltip="Message is mandatory"
      type="button"
    >
      <i
        class="tlp-button-icon fas fa-save"
      />
       
      <span
        data-current-language="en_US"
        data-msgid="Save the configuration"
      >
        Save the configuration
      </span>
    </button>
  </div>
</div>
`;
